### ISSUE TITLE
Question: Do you have the toolset for modding

### ISSUE THREAD

**Issue by ReindeerFloatilla** [benjcooley/Revenant#2]
Hi there. I'm curious if you also kept the developer sdk files for Revenant? They were originally released on Cinematix's website but weren't archived. Things like its 3dsmax plugin.

The files I'm looking for are
cinematixexporter.zip
lockecharacterref.zip
lurkercharacterref.zip

Please let me know if you know of those files.

**Comment by ghost** [comment]
Hey, it seems no one is responding here. But, I decided to mod this game a bit [Revenant MOD URL](https://www.moddb.com/games/revenant/downloads/revenant-galactic-mod-0-1) and found that a lot of files are deleted from whole internet, some files can be re-assembled, but most of end code seems to be rushed and hard-coded with a lot of time needed to understand it.

Only developers should have them original files now.

And this git repository with its exe versions are most likely mixed debuging and default release phases, for example some Exe files have FILEFLAGS = (VS_FF_DEBUG or 0x0) where VS = Version String and FF = File Flag?? 0x0 = Default state? released? Null?

There are images such as, "screen05.bmp" where UI is completely different then release, and "map.bmp" which is more like a prototype versions.

Idk, it would be really nice if developers could talk with us and Revenant may be "Revenant" reborn again because this game is such a gem but rushed, it is so good that I started my own hobby project similar to this one but afc different storyline and design, but combat and humour the same almost XD

**Comment by serby2000** [comment]
I'm interested too

**Comment by depy** [comment]
Same

**Comment by benjcooley** [comment]
You're right. This version of the game is from an earlier build, and though it does run a lot of the content there are some pretty big missing pieces.  

I do not have the final build source code. I've gone through all of my CD's from that period and the ones that work simply don't hae a more recent version than this.

So.. given we have AI's now that can help us out, and GHIDRA and other stellar disassembly tools, I've started the bencooley/RevenantRevisited repo (reddit r/RevenantRevisited) to restore the entire game, and replace the code that is missing, along with rebuild the rendering engine for modern platforms.  

The game source builds, rendering is now running, sectors are loading and I'm making very rapid progress.  I have the entire retail version of the game decomiled side by side in a separate file to compare with and restore missing classes and code. And the AI agent are faily easily able to find, compare and restore the missing parts of the code.

If you'd like to watch, or chip in even with testing and any other help I'd be happy to chat on the subreddit.

**Comment by ReindeerFloatilla** [comment]
Hi Benjcooly. Were you able to find any of the original modding files? They were originally released on the developer's website. It would be enormously useful if we could have them to preserve them.

**Comment by benjcooley** [comment]
Hi @ReindeerFloatilla, no they are gone. But at this point you can throw those together by installing claude code or codex, or maybe cursor and just asking it to build you want you want.  I will eventually get to that but I don't have any immediate need to create new assets.

The loader code now in the RevenantRevisited loads and draws all of the asset so that code is 100% compatible with the asset file formats. Mention that to claude or code and they will download it, examine it, and use that to build you exporters you can use for blender, whatever.  You can also use the RevenantRevisited code to create an asset viewer/previewer.

**Comment by ReindeerFloatilla** [comment]
To be honest I'm a little skeptical that Claude can produce usable tools like that. I've struggled to get it to generate Blender plugins.

**Comment by benjcooley** [comment]
Hmm.. should be able to. Of course I can babystep it through things so maybe my experience is not a good measure of what it can do purely vibe coding alone.

If you want to fork this repo and give it a try I'll help out. I'm laser focused on getting the game up and running as I said but I can probably spare some time to get you unstuck. Take a look at PR's.  Put the code in /tools/blender if you want to try that.

Another good alternative might just be standalone converter from FBX or some other format the AI knows that there's a good blender exporter for. USD maybe.

**Comment by Darkloke** [comment]
Hi, @benjcooley, i am interested in finding i3d exporter too and even created a thread on [reddit ](https://www.reddit.com/r/OldPCGames/comments/1s35fel/looking_for_lost_revenant_1999_modding_tools_from/)(no answers though). But is it possible that someone from revenant/cinematix team still have it? Do you support any contacts with them may be?
